nifi-extension-bundles/nifi-kafka-bundle/nifi-kafka-3-service/src/main/java/org/apache/nifi/kafka/service/Kafka3ConnectionService.java

```diff
@@ -146,6 +149,22 @@ public class Kafka3ConnectionService extends AbstractControllerService implement
             )
             .build();
 
+    public static final PropertyDescriptor KERBEROS_CREDENTIALS_SERVICE = new PropertyDescriptor.Builder()
+            .name("kerberos-credentials-service")
+            .displayName("Kerberos Credentials Service")
+            .description("Service supporting generalized credentials authentication with Kerberos")
+            .identifiesControllerService(KerberosUserService.class)
+            .required(false)
+            .build();
```

Comment:
It has been a while since I worked on this, but I am not sure the Kafka processors can use the regular `KerberosUserService` because they expect to get the JAAS config string from the service and give it to the Kafka client which does the actual login, and this won't work for some of the implementations of `KerberosUserService`, this was the reason for limiting it to the `SelfContainedKerberosUserService`, see comment here:

https://github.com/apache/nifi/blob/main/nifi-extension-bundles/nifi-standard-services/nifi-kerberos-user-service-api/src/main/java/org/apache/nifi/kerberos/SelfContainedKerberosUserService.java#L20-L25

Reply:
Thanks for confirming.  All this lines up perfectly with my observations.  I didn't try `KERBEROS_CREDENTIALS_SERVICE`; I'll remove it.  I was able to get things working with `KERBEROS_USER_SERVICE`, which referenced `SelfContainedKerberosUserService`, by adding the custom property `sasl.jaas.config` as described in this Confluent documentation:

https://docs.confluent.io/platform/7.3/kafka/authentication_sasl/authentication_sasl_gssapi.html#kafka-sasl-auth-gssapi

Also assuming that `KERBEROS_SERVICE_NAME` is needed, as it would seem that NiFi would need a means to convey which service it wants to authenticate to.